### PR TITLE
Defining new OpenT2TError class in the library

### PR DESCRIPTION
**Pull request by shalinijoshi19**
- Simple class that extends built-in Error interface.
- Expected to be referenced in Translators + HubController + consumers of HubController

### CODE REVIEW

**Comment by jasongin** [comment]
LGTM.
Make sure you run the linter before submitting, as I see some minor formatting issues.
```
npm run lint
```

**Comment by shalinijoshi19** [comment]
Thanks a ton.. Will do and will update

Sent from my Windows Phone
________________________________

**Comment by shalinijoshi19** [comment]
Done! Fixed a bunch of linter errors; 
The only ones that i wasnt sure about fixing (yet) were these in existing code: 

lib/JsonSchema.ts[3, 20]: require statement not part of an import statement
lib/JsonSchema.ts[64, 13]: Expected method shorthand in object literal.
lib/JsonSchema.ts[64, 19]: non-arrow functions are forbidden
lib/OpenT2T.ts[173, 33]: non-arrow functions are forbidden
lib/package/LocalPackageSource.ts[364, 17]: Expected property shorthand in object literal.
lib/package/PackageInfo.ts[43, 21]: Expected property shorthand in object literal.
lib/package/PackageInfo.ts[87, 21]: Expected property shorthand in object literal.
lib/package/PackageInfo.ts[89, 21]: Expected property shorthand in object literal.
lib/package/PackageInfo.ts[90, 21]: Expected property shorthand in object literal.
lib/package/PackageInfo.ts[100, 13]: Expected property shorthand in object literal.
lib/package/PackageInfo.ts[101, 13]: Expected property shorthand in object literal.
lib/schema/AllJoynSchemaReader.ts[156, 13]: Expected property shorthand in object literal.
lib/schema/AllJoynSchemaReader.ts[158, 13]: Expected property shorthand in object literal.
lib/schema/AllJoynSchemaReader.ts[178, 13]: Expected property shorthand in object literal.
lib/schema/AllJoynSchemaReader.ts[219, 13]: Expected property shorthand in object literal.
lib/schema/AllJoynSchemaReader.ts[249, 13]: Expected property shorthand in object literal.
lib/schema/AllJoynSchemaReader.ts[250, 13]: Expected property shorthand in object literal.
lib/schema/OcfSchemaReader.ts[144, 21]: Expected property shorthand in object literal.
lib/schema/OcfSchemaReader.ts[150, 13]: Expected property shorthand in object literal.
lib/schema/OcfSchemaReader.ts[151, 13]: Expected property shorthand in object literal.


The shorthand property definition i wasnt sure if we wanted to change (this was much more descriptive i thought) ; With respect to teh non-arrow functions error i'll have to sync with you on how best to tackle.

Also would be great to have this support for other repos -- talking with Chuck to see what it would take to enable the linter for js on other repos



---
In reply to: [279591716](https://github.com/openT2T/opent2t/pull/31#issuecomment-279591716) [](ancestors = 279591716)

**Comment by ChuckFerring-zz** [comment]
:shipit:

File: node/lib/JsonSchema.ts [hunk 1]
```diff
@@ -67,17 +67,17 @@ export class JsonSchema {
                 // (The top schema is probably not in the CWD.)
                 let relativeUri: string = path.relative(process.cwd(), file.url);
                 return fileResolver(relativeUri);
-            }
+            },
         };
 
         let dereferencedSchema: JsonSchema = await $RefParser.dereference(jsonSchema, {
+            dereference: {
```
Comment (shalinijoshi19) [inline node/lib/JsonSchema.ts:74]: linter complaining about alphabetical order - hence the fix #Resolved

File: node/lib/OpenT2TError.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+
+/**
+ * Custom Error class that extends built-in Error.
+ */
+export class OpenT2TError extends Error {
+    public statusCode: number;
```
Comment (kurstory) [inline node/lib/OpenT2TError.ts:7]: To clarify, Status Code uses the same mapping as Http Status Codes?

Comment (shalinijoshi19) [reply node/lib/OpenT2TError.ts:7]: Correct

File: node/lib/OpenT2TConstants.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+
+/**
+ * Custom Error class that extends built-in Error.
+ */
+export class OpenT2TConstants {
+    // Device Names
+    public static WinkHubName = "Wink Hub";
```
Comment (arjun-msft) [inline node/lib/OpenT2TConstants.ts:7]: >public static WinkHubName = "Wink Hub"; [](start = 4, length = 39)

why do we need this here?

**Comment by arjun-msft** [comment]
:shipit:

**Pull request merged**